src/hooks/use-ordered-rows.ts

```diff
@@ -2,9 +2,12 @@ import { useMemo } from 'preact/hooks';
 
 import type { Order } from '../types';
 
+const collator = new Intl.Collator(undefined, { sensitivity: 'case' });
+
 /**
  * Orders a list of rows based on provided order options.
  * Provided rows are not mutated, but a copy is returned instead.
+ * Values are compared preserving their type.
```

Comment:
I would say something like, "strings are compared using `Intl.Collator`, other types are compared using standard JavaScript comparisons". The reader can refer to the `Intl.Collator` docs to understand the fine details.